apps/wallpaper/js/share.js

```diff
@@ -70,8 +73,8 @@ window.onload = function() {
 
         // Compute a scale that will make the image at least as big as
         // the screen in both dimensions.
-        var scalex = window.innerWidth / previewImage.width;
-        var scaley = window.innerHeight / previewImage.height;
+        var scalex = screen.width / previewImage.width;
+        var scaley = screen.height / previewImage.height;
```

Comment:
I'm not sure about changing window.innerWidth to screen.width here and elsewhere.

IIRC, the reason we use screen.width above is that when there is a soft home button (on nexus 4, e.g.) screen.height and window.innerHeight have different values. But we still want the wallpaper to be the full screen height.  I think that the rest of the computations are for manipulating the wallpaper preview image, and for those, using the window dimensions instead of the screen dimensions is correct.

I don't remember the details clearly. If you do a git blame on line 30, though, you might be able to figure out what bug it was when we started using screen dimensions instead of window dimensions, and that should explain why we're doing it.